File: chatgpt-generated/python/xxe.py
import xml.etree.ElementTree as ET
from flask import Flask, request

app = Flask(__name__)

@app.route("/upload", methods=["POST"])
def upload():
    xml_data = request.data
    try:
        root = ET.fromstring(xml_data)  # XXE moguć ovde
        return f"Root tag: {root.tag}"
    except Exception as e:
        return f"Error parsing XML: {e}", 400

if __name__ == "__main__":
    app.run(debug=True)

File: chatgpt-generated/go/xxe.go
package main

import (
	"encoding/xml"
	"fmt"
	"io"
	"net/http"
)

type Root struct {
	XMLName xml.Name
	Content string `xml:",innerxml"`
}

func main() {
	http.HandleFunc("/upload", func(w http.ResponseWriter, r *http.Request) {
		var root Root
		data, _ := io.ReadAll(r.Body)
		xml.Unmarshal(data, &root)
		fmt.Fprintf(w, "Parsed content: %s", root.Content)
	})
	http.ListenAndServe(":8080", nil)
}
